errgroup.go

```diff
@@ -154,13 +156,28 @@ func (g *Group) Go(f func() error) {
 		return
 	}
 
-	g.qCh <- f
+	for {
```

Comment:
In this for loop we simply do the following:
- Get a read lock on err
- If err is NOT nil we return as there is no need to send function to buffer
- If err is nil we try to send to channel, if we can't send to channel then we break the select and go back to checking if err is not nil

Reply:
The only concern with this is it could become a busy loop. I wonder if `sync.Cond` can do something better here